packages/components/forms/stories/FormControl.stories.tsx

```diff
@@ -13,12 +21,74 @@ export default {
 
 export const Basic = (args: FormControlInternalProps) => {
   return (
-    <FormControl {...args}>
-      <FormControl.Label isRequired>Name</FormControl.Label>
-      <TextInput label="input" />
+    <>
+      <FormControl {...args}>
+        <FormControl.Label isRequired>Name</FormControl.Label>
+        <TextInput label="input" />
+        <FormControl.HelpText>
+          Please enter your first name
+        </FormControl.HelpText>
+        {args.isInvalid && (
+          <FormControl.ValidationMessage>Error</FormControl.ValidationMessage>
+        )}
+      </FormControl>
+
+      <FormControl {...args}>
+        <FormControl.Label>Description</FormControl.Label>
+        <Textarea label="input" />
+        <FormControl.HelpText>Tell me about youself</FormControl.HelpText>
+        {args.isInvalid && (
+          <FormControl.ValidationMessage>Error</FormControl.ValidationMessage>
+        )}
+      </FormControl>
+    </>
+  );
+};
+
+export const Invalid = (args: FormControlInternalProps) => {
+  return (
+    <Basic {...args} isInvalid>
+      {args.children}
+    </Basic>
+  );
+};
+
+export const WithCustomLogic = (args: FormControlInternalProps) => {
+  const [isDisabled, setIsDisabled] = React.useState(true);
+  return (
+    <FormControl {...args} isDisabled={isDisabled}>
+      <Flex justifyContent="space-between" alignItems="center">
+        <FormControl.Label isRequired>Name</FormControl.Label>
+        <Box marginBottom="spacingS">
+          {isDisabled && (
+            <TextLink
+              as="button"
+              icon={<LockIcon />}
+              onClick={() => {
+                setIsDisabled(false);
+              }}
+            >
+              Unlock to edit
+            </TextLink>
+          )}
+          {!isDisabled && (
+            <TextLink
+              as="button"
+              icon={<LockIcon />}
+              onClick={() => {
+                setIsDisabled(true);
+              }}
+            >
+              Lock
+            </TextLink>
+          )}
```

Comment:
```suggestion
            <TextLink
              as="button"
              icon={<LockIcon />}
              onClick={() => {
                setIsDisabled(prevState => !prevState);
              }}
            >
              {isDisabled ? 'Unlock to edit' : 'Lock'}
            </TextLink>
          )}
```

I think it's possible to do the same without repeating the code as much, wdyt?

P.S.: it also teaches people about prevState

Reply:
yeah, it's a great suggesion